packages/react-components/react-charts-preview/library/src/utilities/utilities.ts

```diff
@@ -300,31 +329,63 @@ export function createStringXAxis(
 ) {
   const {
     domainNRangeValues,
-    xAxisCount = 6,
     xAxistickSize = 6,
     tickPadding = 10,
     xAxisPadding = 0.1,
     xAxisInnerPadding,
     xAxisOuterPadding,
+    containerWidth,
+    hideTickOverlap,
   } = xAxisParams;
   const xAxisScale = d3ScaleBand()
     .domain(dataset!)
     .range([domainNRangeValues.rStartValue, domainNRangeValues.rEndValue])
     .paddingInner(typeof xAxisInnerPadding !== 'undefined' ? xAxisInnerPadding : xAxisPadding)
     .paddingOuter(typeof xAxisOuterPadding !== 'undefined' ? xAxisOuterPadding : xAxisPadding);
+
+  let tickValues = dataset;
+  const tickFormat = (domainValue: string, _index: number) => {
+    return convertToLocaleString(domainValue, culture) as string;
+  };
+  if (hideTickOverlap) {
+    tickValues = [];
+    const tickSizes = dataset.map((value, index) =>
+      calculateLongestLabelWidth([tickFormat(value, index)], '.fui-cart__xAxis text'),
+    );
+    // for LTR
+    let start = 0;
+    let end = containerWidth;
+    let sign = 1;
+    const range = xAxisScale.range();
+    if (range[1] - range[0] < 0) {
+      // for RTL
+      start = containerWidth;
+      end = 0;
+      sign = -1;
+    }
+    for (let i = dataset.length - 1; i >= 0; i--) {
+      const tickPosition = xAxisScale(dataset[i])!;
+      if (
+        sign * (tickPosition - (sign * tickSizes[i]) / 2 - start) >= 0 &&
+        sign * (tickPosition + (sign * tickSizes[i]) / 2 - end) <= 0
+      ) {
+        tickValues.push(dataset[i]);
+        end = tickPosition - sign * (tickSizes[i] / 2 + 10);
+      }
+    }
+    tickValues = tickValues.reverse();
+  }
+
   const xAxis = d3AxisBottom(xAxisScale)
     .tickSize(xAxistickSize)
     .tickPadding(tickPadding)
-    .ticks(xAxisCount)
-    .tickFormat((x: string, index: number) => {
-      return convertToLocaleString(dataset[index], culture) as string;
-    });
+    .tickValues(tickValues)
+    .tickFormat(tickFormat);
```

Comment:
ToDo: add unit tests for this logic

Reply:
Yes, will add the unit tests once the logic is finalized. Still have to cover scenarios like rotation, wrapping and truncation of tick labels. Also noticed that the calculateLongestLabelWidth util is not very accurate.